### PR TITLE
Fixed _save_model not working for batches in ModelCheckpoint Callback

### LINKED ISSUE #38668: In ModelCheckpoint, filepath is not accepting batch as formatting parameter.
In `ModelCheckpoint` callback, there is a parameter given named `save_freq` to save model. If `save_freq` is set to `epoch`, it will save model at the end of every epoch. (This works perfectly fine). But when `save_freq` is set to an integer let's say `N`, then the callback should save the model after `N` batches in every epoch. But the problem here is the callback doesn't accept the filepath as `file.batch{batch:02d}epoch{epoch:02d}.h5` and raises error as `batch` is invalid key.  
The problem in the code that I have noticed is that the `_save_model` function has access to `epoch` but it doesn't have access to `batch`. And that's why `_get_file_path()` has access to `epoch` but not `batch`.  The functionality should be changed little bit. I am raising PR to add access to `batch` param in both `_save_model` and `_get_file_path` variable.
I noticed this error in tf code during the work on my PR [#1702](https://github.com/tensorflow/addons/pull/1702) in tensorflow/addons.  

cc @gabrieldemarmiesse.

### PR DESCRIPTION
**Pull request by ashutosh1919**
Fixes #38668 .  
@mihaimaruseac - Please review.

### CODE REVIEW

**Review by mihaimaruseac — DISMISSED** [review]
Might need to get approval from API owners too

**Comment by ashutosh1919** [comment]
@mihaimaruseac , sorry. Had two spaces after comma instead of one. Pylint was failing. Please, can you approve it again?

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by ashutosh1919** [comment]
@mihaimaruseac , resolved merge conflict. Please approve.

**Comment by mihaimaruseac** [comment]
Let's wait for TF API owners to review the change in behavior

**Comment by jsimsa** [comment]
(for tensorflow/api-owners)

Deferring to @rchao to review the behavior change.

File: tensorflow/python/keras/callbacks.py
Comment (rchao) [inline tensorflow/python/keras/callbacks.py:1307]: We shouldn't need this if-else anymore since distributed_file_utils.write_filepath would take care of that writing path. Can you fix this?

Comment (ashutosh1919) [reply tensorflow/python/keras/callbacks.py]: Done

**Comment by ashutosh1919** [comment]
@rchao and @mihaimaruseac , I have modified code as per the last review. Please can you review it?

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/python/keras/callbacks.py
Comment (rchao) [inline tensorflow/python/keras/callbacks.py:1341]: Can this fit in one line?

File: tensorflow/python/keras/callbacks.py
Comment (rchao) [inline tensorflow/python/keras/callbacks.py:1281]: Can this fit in the previous line?

File: tensorflow/python/keras/callbacks.py
Comment (ashutosh1919) [reply tensorflow/python/keras/callbacks.py]: No, will give pylint error. But I have changed it to resolve error.

File: tensorflow/python/keras/callbacks.py
Comment (ashutosh1919) [reply tensorflow/python/keras/callbacks.py:1281]: No, will give pylint error. But I have changed it to resolve error.

**Comment by ashutosh1919** [comment]
@rchao , I have tested pylint locally. Errors are resolved now.

**Review by rchao — DISMISSED** [review]
Thanks for the PR!

**Comment by rthadur** [comment]
@rchao is api review done for this change ?

**Comment by mihaimaruseac** [comment]
Seems this failed some tests internally and failed to land. Testing again and will update.

**Comment by mihaimaruseac** [comment]
`//tensorflow/python/keras:callbacks_test` is failing

```
...
File "tensorflow/python/keras/callbacks.py", line 1347, in _get_file_path
    epoch=epoch + 1, batch=batch + 1, **logs)
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Test failure

**Comment by ashutosh1919** [comment]
@mihaimaruseac , Please paste error stack here.

**Comment by mihaimaruseac** [comment]
Apologies, last line of stack trace (with the error) didn't copy above. Here's the full trace

```
Traceback (most recent call last):
  File "<embedded stdlib>/unittest/case.py", line 59, in testPartExecutor
    yield
  File "<embedded stdlib>/unittest/case.py", line 605, in run
    testMethod()
  File ".../absl/testing/parameterized.py", line 284, in bound_param_test
    return test_method(self, *testcase_params)
  File ".../tensorflow/python/keras/keras_parameterized.py", line 280, in decorated
    _test_sequential_model_type(f, self, *args, **kwargs)
  File ".../tensorflow/python/keras/keras_parameterized.py", line 300, in _test_sequential_model_type
    f(test_or_class, *args, **kwargs)
  File ".../tensorflow/python/keras/callbacks_test.py", line 633, in test_ModelCheckpoint
    verbose=1)
  File ".../tensorflow/python/keras/engine/training_v1.py", line 807, in fit
    use_multiprocessing=use_multiprocessing)
  File ".../tensorflow/python/keras/engine/training_arrays.py", line 666, in fit
    steps_name='steps_per_epoch')
  File ".../tensorflow/python/keras/engine/training_arrays.py", line 397, in model_iteration
    callbacks._call_batch_hook(mode, 'end', batch_index, batch_logs)
  File ".../tensorflow/python/keras/callbacks.py", line 284, in _call_batch_hook
    self._call_batch_end_hook(mode, batch, logs)
  File ".../tensorflow/python/keras/callbacks.py", line 304, in _call_batch_end_hook
    self._call_batch_hook_helper(hook_name, batch, logs)
  File ".../tensorflow/python/keras/callbacks.py", line 337, in _call_batch_hook_helper
    hook(batch, logs)
  File ".../tensorflow/python/keras/callbacks.py", line 1243, in on_train_batch_end
    self._save_model(epoch=self._current_epoch, batch=batch, logs=logs)
  File ".../tensorflow/python/keras/callbacks.py", line 1297, in _save_model
    filepath = self._get_file_path(epoch, batch, logs)
  File ".../tensorflow/python/keras/callbacks.py", line 1347, in _get_file_path
    epoch=epoch + 1, batch=batch + 1, **logs)
TypeError: format() got multiple values for keyword argument 'batch'
```

**Comment by jsimsa** [comment]
For @tensorflow/api-owners: no need for API review.

**Comment by gbaned** [comment]
@ashutosh1919 Can you please resolve conflicts? Thanks!

**Comment by mihaimaruseac** [comment]
Ping on resolving conflicts and test failure?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by mihaimaruseac** [comment]
@ashutosh1919 gentle ping

**Comment by ashutosh1919** [comment]
@mihaimaruseac and @rchao - Sorry for the delay. I was very much impacted by COVID-19 and had to stop contributions to tf due to it.

I have resolved merge conflicts. Please run the tests.

**Review by mihaimaruseac — APPROVED** [review]

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
broken tests were not fixed yet

**Comment by gbaned** [comment]
@ashutosh1919, @mihaimaruseac Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
@ashutosh1919, @mihaimaruseac Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
@ashutosh1919, @mihaimaruseac Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
@ashutosh1919 Any update on this PR? Please. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 29 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Pull request closed without merging**
